tests/unit/test_signature.cpp

```diff
@@ -0,0 +1,427 @@
+                _testMessageDigestSHA256));
+
+    EXPECT_NO_THROW(
+          SignatureUtils::verify(mococrw::AsymmetricKeypair::readPublicKeyFromPEM(_validPublicKey),
+                                 signature,
+                                 _testMessageDigestSHA256));
+}
+
+/**
+ * @brief Unsuccessful cases of wrong parameters defined in context for verification
+ * Cases:
+ * - Wrong padding mode assigned to verification context
+ * - Wrong public key assigned to verification context
+ * - Wrong padding assigned to verification context
+ * - Wrong hashing function assigned to verification context
+ * - Wrong masking assigned to verification context
+ * - Wrong salt lengthassigned to verification context
+ */
+TEST_F(SignatureTest, testSignatureUnsuccessfulVerificationCases)
+{
+    std::vector<uint8_t> signature;
+    SignatureUtils::Context<AsymmetricPrivateKey> signatureContext(_keyPair);
+    signatureContext.setPadding(openssl::RSAPaddingMode::PSS);
+    signature = SignatureUtils::create(signatureContext, _testMessage);
+
+    SignatureUtils::Context<AsymmetricPublicKey> verificationContext(*_root1_pubkey);
+    signatureContext.setPadding(openssl::RSAPaddingMode::PSS);
+
+    // Using different public key
+    ASSERT_THROW(SignatureUtils::verify(verificationContext, signature, _testMessage),
+                 MoCOCrWException);
+
+    // Verification context considering different padding mode is set.
+    verificationContext.setKey(_keyPair);
+    verificationContext.setPadding(openssl::RSAPaddingMode::PKCS1);
+    ASSERT_THROW(SignatureUtils::verify(verificationContext, signature, _testMessage),
+                 MoCOCrWException);
+
+    // Verification context considering different hashing is set.
+    verificationContext.setPadding(openssl::RSAPaddingMode::PSS);
+    verificationContext.setHashingFunction(openssl::DigestTypes::SHA512);
+    ASSERT_THROW(SignatureUtils::verify(verificationContext, signature, _testMessage),
+                 MoCOCrWException);
+
+    // Verification context considering different masking is set.
+    verificationContext.setHashingFunction(openssl::DigestTypes::SHA256);
+    verificationContext.setMaskingFunction(openssl::DigestTypes::SHA512);
+    ASSERT_THROW(SignatureUtils::verify(verificationContext, signature, _testMessage),
+                 MoCOCrWException);
+
+    // Verification context considering different salt length is set.
+    verificationContext.setMaskingFunction(openssl::DigestTypes::SHA256);
+    verificationContext.setSaltLength(2);
+    ASSERT_THROW(SignatureUtils::verify(verificationContext, signature, _testMessage),
+                 MoCOCrWException);
+}
+
+// Signature Failure cases
+
+/**
+ * @brief Unsuccessful case where the message digest length is different of padding mode,
+ *        considering the usage of PKCS1.
+ */
+TEST_F(SignatureTest, testSignatureInvalidDigestLengthPKCS1)
+{
+    std::vector<uint8_t> signature;
+    std::vector<uint8_t> messageDigest;
```

Comment:
Mater of taste, but declaring variables way before using them looks very C-ish. (also in other places)